plugins/auto-sizes/hooks.php

```diff
@@ -206,6 +206,14 @@ function auto_sizes_improve_image_sizes_attributes( string $content ): string {
 
 	if ( $sizes ) {
 		$processor->set_attribute( 'sizes', $sizes );
+
+		// The sizes filter to reflate improve sizes calculations.
+		add_filter(
+			'wp_calculate_image_sizes',
+			static function () use ( $sizes ): string {
+				return $sizes;
+			}
+		);
```

Comment:
I don't understand why this filter is being added without being removed. Every time that `wp_filter_content_tags()` is called, another filter `wp_calculate_image_sizes` will be added. And since it is a closure here, the `$sizes` value from one image could end up getting applied to another image unexpectedly?

Reply:
Nice catch. Added remove filter code.